src/hooks/useGiveInfo.ts

```diff
@@ -35,21 +50,31 @@ export const useDonationInfo = () => {
     async () => {
       queryAssertion([address, networkId], donationInfoQueryKey(address, networkId));
 
+      // Set default return value
       const donationInfo: IUserDonationInfo[] = [];
 
-      if (!contract) return donationInfo;
+      // If there's no contract (i.e. on a non-ETH network), throw error
+      if (!contract)
```

Comment:
How does the frontend handle throwing this error? Please test that. Does it cause the frontend to crash? Probably a bad thing. Perhaps it should call `dispatch(error("etc..."))`? But the calling function should handle this accordingly.

Reply:
It just reports it as an error in console like this. what do you think?: 
<img width="1710" alt="image" src="https://user-images.githubusercontent.com/92545857/162282949-55dbf70c-874a-4c14-bcdf-8f1cf32d2edf.png">

Reply:
I think it'll also show an error message in the interface (based on my experience with it in OlyZaps), which is fine.